core/lib/libtomcrypt/src/ciphers/sub.mk

```diff
@@ -1,5 +1,13 @@
 cflags-y += -Wno-unused-parameter
 
+ifeq ($(CFG_CRYPTO_AES_ARM64_CE),y)
+srcs-y += aes_arm64_ce.c
+cflags-aes_arm64_ce.c-y += -march=armv8-a+crypto
```

Comment:
Yes, the possible use of SIMD registers by unrelated compiler generated code is a concern.
In Linux, this is addressed implicitly since the entire arm64 kernel is built with -mgeneral-regs-only. In OP-TEE you may want to add this option as well to make 100% sure that the FP/SIMD is never touched (+simd is the default afaik)